interactivity/configuring-ai-powered-insights.md

```diff
@@ -0,0 +1,428 @@
+            "allowCustomPrompts": false,
+            "predefinedPrompts": [
+                { "text": "Generate a summary of the report." },
+                { "text": "Translate the report into German." }
+            ],
+        }
+    }
+}
+````
+````XML
+<Telerik.Reporting>
+    <AIClient
+        <!-- ...base configuration... -->
+        requireConsent="false"
+        allowCustomPrompts="false">
+        <predefinedPrompts>
+            <add text="Generate a summary of the report." />
+            <add text="Translate the report into German." />
+        </predefinedPrompts>
+    </AIClient>
+</Telerik.Reporting>
+````
+
+You can also add predefined prompts without disabling custom ones, giving users both curated options and the flexibility to create their own queries.
+
+### Retrieval-Augmented Generation (RAG) Configuration
+
+By default, the AI-powered insights functionality uses a [Retrieval-Augmented Generation (RAG)](https://aws.amazon.com/what-is/retrieval-augmented-generation/) algorithm to filter out the irrelevant report data before sending it to the AI model. This approach significantly improves the accuracy and relevance of the AI-generated response while optimizing token usage.
+
+> RAG is available only in .NET and .NET Standard. Therefore, the options that are listed below are not supported in .NET Framework configurations.
+
+If needed, you can disable this algorithm by setting `allowRAG` to `false`.
+
+You can also configure the RAG behavior through the `ragSettings` option:
+- `modelMaxInputTokenLimit`&mdash;Limits the maximum input tokens the AI model can process in a single request. The default value is `15000`.
+- `maxNumberOfEmbeddingsSent`&mdash;Limits how many embeddings (chunks of retrieved content) are sent to the model in a single request. The default value is `15`.
+- `maxTokenSizeOfSingleEmbedding`&mdash;Limits token size of each individual embedding, which prevents large chunks from dominating the prompt. The default value is `0` (no limit).
+- `tokenizationEncoding`&mdash;Specifies tokenization scheme used to estimate the tokens usage before sending the request to the LLM model. By default, the encoding is determined automatically based on the specified model, which is recommended to ensure accurate token counting. Incorrect encoding may lead to miscalculations in token limits, causing either premature truncation of context or exceeding the model’s input capacity.
+- `splitTables`&mdash;Indicates whether tables should be split during Retrieval-Augmented Generation (RAG) processing. When the splitting is allowed, only the relevant table cells will be taken into account, significantly reducing the number of tokens. The default value is `true`.
+
+Below is an example that takes advantage of the table splitting and automatic encoding inference, but reduces the token limits:
+
+````JSON
+"telerikReporting": {
+	"AIClient": {
+        // ...base configuration...
+		"requireConsent": false,
```

Comment:
Again, maybe skip the irrelevant settings: requireConsent, allowCustomPrompts, predefinedPrompts. This would avoid confusion that these are interdependent, both for human readers and for our coding assistant.